lib/interactive_logger/step.rb

```diff
@@ -60,9 +61,20 @@ def prefix(str)
     end
 
     def print_msg(str)
+      str = terminal_trim(str)
       blank
       @last_print_msg = str
       print str
     end
+
+    # Trim string to current terminal width and break at newline so it can be
+    # replaced rather than causing the same line to print multiple times.
+    def terminal_trim(str)
+      terminal_width = @thor_shell.terminal_width
```

Comment:
Looks like Ruby 1.9.2 added native support for this:
```ruby
2.1.6 :005 > IO.console.winsize
 => [27, 125]
```

I'm working on a fix to highly active threaded step debug info using mutexes and may just merge your patch in using this.